kontent-ai/delivery-sdk-net
# Generic Filters

### Motivation

Fixes #119

I did not add the specific `DateTime` and `decimal` constructors. I have also kept the original method signatures for all of the filters for backward compatibility.
The issue (#119) calls out that all filters should implement `DateTime` and `decimal` constructors, then utilize a new constructor in `Filter` to accept `object`s as values. The original issue (#27), has a proposed solution in such a way; however, doing so defeats the purpose of creating a generic filter in the first place. Since, if we were to go the `object` route, all values would get converted to a `string`, removing the need for a generic and for `GetQueryStringParameter` to handle types in a `switch`.
Instead, I opted to go the route of the generic filter and implemented a switch statement to interpret `string`, `DateTime`, and numeric types (`int`, `decimal`, `float`, etc.). The downside here is that we lose the specific constructors for `DateTime` and `decimal` values.

### Checklist


### How to test

Create queries using the new filters and confirm that results are correctly filtered.
